server/game/core/Player.js

```diff
@@ -965,6 +965,25 @@ class Player extends GameObject {
         }
     }
 
+    /**
+     * If possible, exhaust the given resource and ready another one instead
+     */
+    swapReadyResource(resource) {
```

Comment:
Maybe `swapResourceReadyState` would be a touch clearer, it took me a second to figure out what this was for